core/camel-support/pom.xml

```diff
@@ -85,4 +85,53 @@
         </dependency>
     </dependencies>
 
+    <profiles>
+        <profile>
+            <id>java-21-sources</id>
```

Comment:
No unfortunately maven doesn't support it out of the box. They only propose different approaches to do so https://maven.apache.org/plugins/maven-compiler-plugin/multirelease.html and I more or less picked one of them.
To remove the profile, I would need to enforce Java 21 but I wanted to avoid being too intrusive but if it is not an issue for you all, I can enforce it.

Reply:
Understood. Thanks for the explanation.